scrapy/extensions/feedexport.py

```diff
@@ -118,6 +118,7 @@ def __init__(self, uri, access_key=None, secret_key=None):
         self.secret_key = u.password or secret_key
         self.is_botocore = is_botocore()
         self.keyname = u.path[1:]  # remove first "/"
+        self.policy = settings.get('FEED_STORAGE_S3_ACL', 'private')
```

Comment:
I think the `settings` variable is only defined in the `if no_defaults` block above

Reply:
and I think it's better to use a default value of `None` instead of private since the default value for `polify` kw arg in boto `set_contents_from_file` is also None

```
set_contents_from_file(fp, headers=None, replace=True, cb=None, num_cb=10, policy=None, md5=None, reduced_redundancy=False, query_args=None, encrypt_key=False, size=None, rewind=False)
```

http://boto.cloudhackers.com/en/latest/ref/s3.html#boto.s3.key.Key.set_contents_from_file

Reply:
> I think the settings variable is only defined in the if no_defaults block above

that's true, but it also means the policy should be passed from `from_crawler` method as constructor argument defaulting to `None`.